src/goto-programs/goto_trace.cpp

```diff
@@ -91,7 +91,7 @@ void goto_trace_stept::output(
 
   out << "\n";
 
-  if(pc->is_other() || pc->is_assign())
+  if((pc->is_other() && lhs_object.is_not_nil())|| pc->is_assign())
```

Comment:
A good commit message would ideally explain why this condition is now needed. This seems fairly unrelated to this PR and ideally this should have a test demonstrating where this didn't work before.

Reply:
Here is a scenario that was failing before but works now: 
``` c++
SCENARIO("trace.output", "[core]")
{
  symbol_tablet symbol_table;
  namespacet ns(symbol_table);
  goto_programt::instructionst instructions;
  instructions.emplace_back(goto_program_instruction_typet::OTHER);
  goto_trace_stept step;
  step.pc=instructions.begin();
  step.type=goto_trace_stept::typet::ATOMIC_BEGIN;
  step.output(ns, std::cout);
}
```

I could add it as a unit test

Reply:
That seems like a good thing to add if that is considered a reasonable trace :)